### PR TITLE
qemu: disable meson compiler check

### PR DESCRIPTION
**Pull request by herbygillot**
###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: emulators/qemu/files/patch-meson-disable-compiler-check.diff [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+-      # if __clang_major__ < 10 || (__clang_major__ == 10 && __clang_minor__ < 0)
+-      #  error You need at least Clang v10.0 to compile QEMU
+-      # endif
+-      #endif''')
+-    # ok
+-  else
+-    error('You either need GCC v7.4 or Clang v10.0 (or XCode Clang v15.0) to compile QEMU')
+-  endif
+-endforeach
++#foreach lang : all_languages
```
Comment (raimue) [inline emulators/qemu/files/patch-meson-disable-compiler-check.diff:26]: Just remove the lines, do not comment them out. This makes it more complex to see what the patch is doing.

**Review by raimue — COMMENTED** [review]
What's the rationale why this check needs to be disabled? Upstream must have this for good reasons.

**Comment by herbygillot** [comment]
> What's the rationale why this check needs to be disabled? Upstream must have this for good reasons.

After updating to the latest version of `qemu`, builds are failing across the board on the buildbots because of this check:
https://build.macports.org/builders/ports-13_x86_64-builder/builds/83793/steps/install-port/logs/stdio

```
../qemu-9.2.0/meson.build:328:4: ERROR: Problem encountered: You either need GCC v7.4 or Clang v10.0 (or XCode Clang v15.0) to compile QEMU

A full log can be found at /opt/local/var/macports/build/_opt_bblocal_var_buildworker_ports_build_ports_emulators_qemu/qemu/work/build/meson-logs/meson-log.txt
```

Wanted to see if this will actually still compile on the compiler versions that are present.

**Comment by ryandesign** [comment]
This is why they increased the compiler requirements:

https://github.com/qemu/qemu/commit/4e035201b961cd545b6801f6656908ecc3055c69

So, no particular reason, just only wanting to support the latest couple versions. It's reasonable to see if older versions still work, within reason: qemu requires macOS 12 or later (they have deliberately removed compatibility code for earlier systems), and that needs to be reflected in the portfile, which will close several tickets.

**Comment by raimue** [comment]
I updated the compiler.blacklist to represent the new requirement with d18f1dda1c5a8e5e0990ac2771db2fbaf7b86828. I think this should be enough to lead to more meaningful error messages for users.

@herbygillot Are you fine that we drop this PR then?

**Comment by herbygillot** [comment]
> I updated the compiler.blacklist to represent the new requirement with [d18f1dd](https://github.com/macports/macports-ports/commit/d18f1dda1c5a8e5e0990ac2771db2fbaf7b86828). I think this should be enough to lead to more meaningful error messages for users.
> 
> @herbygillot Are you fine that we drop this PR then?

I suppose so.

I'm wondering if we should peg older versions of macOS to earlier versions of qemu that would still work for those macOS versions....

**Pull request closed without merging**